actors/builtin/miner/miner_state.go

```diff
@@ -102,8 +102,11 @@ type MinerInfo struct {
 	// Slice of byte arrays representing Libp2p multi-addresses used for establishing a connection with this miner.
 	Multiaddrs []abi.Multiaddrs
 
-	// The proof type used by this miner for sealing sectors.
-	SealProofType abi.RegisteredSealProof
+	// The proof types used for PoSt for this miner.
+	// A miner may commit sectors with different seal proof types (but compatible sector size and
+	// corresponding PoSt proof types).
+	WindowPoStProofType  abi.RegisteredPoStProof
+	WinningPoStProofType abi.RegisteredPoStProof
```

Comment:
@nicola you have previously noted that changing schemas like this makes dynamic interpretation (e.g. of JSON versions) across versions by external tools more difficult. To retain semantic compatibility here, we would need to leave a dummy integer field here, then append these two new ones at the end of the struct. I don't that's a huge pain or cost, so am willing to do so. 

We won't always get off so easily though - is it worth it? Tooling in general will need to understand these schemas and use the right one. Feedback also welcome @frrist @iand @ribasushi @willscott

Reply:
@rvagg would be the best person to weigh in from IPLD on the above

Reply:
It does break my tooling that does not have any of the above, since they assumed that things like this would not happen.

However, with good engineering on libraries and clear schemas it shouldn't be a problem. The expert here is @willscott.

Reply:
Array length helps differentiate these changes, we're adding +1 to this array and there's no previous form with this many - [v0 has 9, v2 has 11](https://specs.ipld.io/data-structures/filecoin/state.html#storagemineractor), this has 12, so this should be as straightforward as other changes that have been introduced.

Reply:
Given that the array length identifies the schema (this time), I'm going to leave it in this natural form, with no dummy.

Reply:
Actually, I'm going to remove the Winning PoSt proof type, which will revert this to 11 fields. I think I'll just rename the proof type, but leave it in the existing array position. Since they were the same serialized type, the schema remains the same. The values will be disjoint between v2 and v3. I think this is likely to be the easiest thing for schema-less consumers to use anyway.